packages/contracts-rfq/contracts/FastBridgeV2.sol

```diff
@@ -335,33 +324,28 @@ contract FastBridgeV2 is Admin, IFastBridgeV2, IFastBridgeV2Errors {
         return abi.decode(request, (BridgeTransactionV2));
     }
 
-    /// @notice Pulls a requested token from the user to the requested recipient.
-    /// @dev Be careful of re-entrancy issues when msg.value > 0 and recipient != address(this)
-    function _pullToken(address recipient, address token, uint256 amount) internal returns (uint256 amountPulled) {
-        if (token != UniversalTokenLib.ETH_ADDRESS) {
+    /// @notice Pulls a requested token from the user to this contract.
+    function _pullToken(address token, uint256 amount) internal returns (uint256 amountPulled) {
+        if (token == UniversalTokenLib.ETH_ADDRESS) {
+            // For ETH we just need to check that the supplied msg.value is correct
+            if (amount != msg.value) revert MsgValueIncorrect();
+            amountPulled = msg.value;
+        } else {
             token.assertIsContract();
             // Record token balance before transfer
-            amountPulled = IERC20(token).balanceOf(recipient);
+            amountPulled = IERC20(token).balanceOf(address(this));
             // Token needs to be pulled only if msg.value is zero
```

Comment:
is this comment right? seems like msg.value can be !=0 and token still pulled if its an ERC20 bridge w/ callValue

Reply:
Good catch - seems that this comment (with no value check) is present in `FastBridge` as well. More revert cases for incorrectly invoked `bridge()` by the users to cover that would be nice, will make a separate ticket for that.

As for the `_pullToken`, it probably needs a rename to signal that it's only used in `bridge()` now.

Reply:
@parodime the comment was fixed to be enforced in #3251, while the function itself got refactored in 189931c968e016c2f9d1f431ad8aebace124f6d7. Does this look cleaner now?